connections/sim/hw/hw_sim.py

```diff
@@ -44,6 +61,7 @@ def digital_write(self, pin, val):
             elif pin in self._ignitor_fires and val:
                 self._ignitor_fires[pin].fire()
 
+
     def analog_read(self, pin):
```

Comment:
In here we should assert that the pin is set as an input.

Reply:
We are now checking to see that the pin is set as an input

bd9f4ee